src/blssignature.cpp

```diff
@@ -36,35 +36,134 @@ BLSSignature BLSSignature::FromBytes(const uint8_t *data) {
     return sigObj;
 }
 
-BLSSignature BLSSignature::FromBytes(const uint8_t *data,
-                                     const AggregationInfo &info) {
+BLSInsecureSignature BLSInsecureSignature::FromG2(const relic::g2_t* element) {
+    BLS::AssertInitialized();
+    BLSInsecureSignature sigObj = BLSInsecureSignature();
+    relic::g2_copy(sigObj.sig, *(relic::g2_t*)element);
+    return sigObj;
+}
+
+BLSInsecureSignature::BLSInsecureSignature() {
+    BLS::AssertInitialized();
+    g2_set_infty(sig);
+}
+
+BLSInsecureSignature::BLSInsecureSignature(const BLSInsecureSignature &signature) {
+    BLS::AssertInitialized();
+    g2_copy(sig, *(relic::g2_t*)&signature.sig);
+}
+
+void BLSInsecureSignature::GetPoint(relic::ep2_st* output) const {
+    *output = *sig;
+}
+
+BLSInsecureSignature BLSInsecureSignature::Aggregate(const BLSInsecureSignature& r) const {
+    BLSInsecureSignature result(*this);
+    g2_add(result.sig, *(relic::g2_t*)&result.sig, *(relic::g2_t*)&r.sig);
+    return result;
+}
+
+BLSInsecureSignature BLSInsecureSignature::Aggregate(const std::vector<BLSInsecureSignature>& sigs) {
+    if (sigs.empty()) {
+        throw std::string("sigs must not be empty");
+    }
+    BLSInsecureSignature result = sigs[0];
+    for (size_t i = 1; i < sigs.size(); i++) {
+        g2_add(result.sig, result.sig, *(relic::g2_t*)&sigs[i].sig);
+    }
+    return result;
+}
+
+BLSInsecureSignature BLSInsecureSignature::DivideBy(const BLSInsecureSignature& r) const {
```

Comment:
We should have divide by vector of signatures as well

Reply:
Actually, let's only have the vector versions of everything for now, to keep API simpler